src/Hosting/Hosting/src/WebHostBuilder.cs

```diff
@@ -179,7 +179,7 @@ public IWebHost Build()
             var assemblyNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
             foreach (var assemblyName in _options.GetFinalHostingStartupAssemblies())
             {
-                if (!assemblyNames.Add(assemblyName))
+                if (!assemblyNames.Add(assemblyName) && logger.IsEnabled(LogLevel.Warning))
                 {
                     logger.LogWarning($"The assembly {assemblyName} was specified multiple times. Hosting startup assemblies should only be specified once.");
```

Comment:
There's an analyzer that says to pass variable data as parameters, right?
```suggestion
                    logger.LogWarning("The assembly {assemblyName} was specified multiple times. Hosting startup assemblies should only be specified once.", assemblyName);
```